app/src/views/ResultPage.vue

```diff
@@ -249,16 +258,26 @@
         methods: {
             ...mapActions(['hydrateStateFromRoute', 'updateURIFromState', 'setSelectedPost', 'setPage', 'findPosts']),
             openPost(id: string): void {
+                        this.postMapToggle = 'post';
                         this.setSelectedPost(this.posts.find((post) => post.id === id));
                     },
             closePost(): void {
                 const currentPost = this.selectedPost as Post;
                 const location = [currentPost.geo_location.lat, currentPost.geo_location.lon] as LatLngTuple;
+                this.postMapToggle = 'map';
                 this.setSelectedPost(null);
                 this.$nextTick(() => {
                     (this.$refs.map as LMap).setCenter(location);
                 });
             },
+            openMap(): void {
+                const currentPost = this.selectedPost as Post;
```

Comment:
This is duplicated code. Please refractor closePost method to call openMap and setSelectedPost to null

Reply:
If you set the selected post to null, then you can't get back to the detail view from the map.

Reply:
but the exact same thing happens in https://github.com/hochschule-darmstadt/einander-helfen/pull/70/files/56524393b2f20adec6ec362434ddf6527a01e054#diff-4511ef840022454a3e8fa7ea0e537489R268

Reply:
Maybe I didn't make myself clear.
Your openMap() method is 6 lines. 
The closePost() method is 7 lines of which 6 are the same as the openMap() method.

Please refactor the closePost() method to utilize the openMap() method to reduce duplicated code.

Reply:
If the current post is not being set to null when closing the post the URI won`t be updated and still reflect the openPost state.

Reply:
I think this is fine because switching to the map doesn't mean you want to close the post entirely. You just want to see the currently selected post on the map and then be able to return to the details page by clicking the "Details" button.  I have now changed the function closePost() to call the function openMap(). Should I delete the function closePost()?